File: src/gestorAplicacion/administracion/HistoriaClinica.java
/* Autores: Diego Andres Gracia Granados, Daniel Giraldo Vanegas, Elian David Velandia Riveros, Juan Camilo Gutierrez Martinez
y Santiago Arboleda Acevedo */
package gestorAplicacion.administracion;

import gestorAplicacion.personas.Doctor;
import gestorAplicacion.personas.Enfermedad;
import gestorAplicacion.personas.Paciente;
import gestorAplicacion.servicios.Cita;
import gestorAplicacion.servicios.CitaVacuna;
import gestorAplicacion.servicios.Formula;

import java.io.Serializable;
import java.util.ArrayList;

//Clase destinada a crear historias clínicas a pacientes
public class HistoriaClinica implements Serializable {
    //Atributos
    private final Paciente paciente;
    private ArrayList<Formula> listaFormulas = new ArrayList<>();
    private ArrayList<Cita> historialCitas = new ArrayList<>();
    private ArrayList<Enfermedad> enfermedades = new ArrayList<>();
    private ArrayList<CitaVacuna> historialVacunas= new ArrayList<>();

    //Constructor
    public HistoriaClinica(Paciente paciente) {
        this.paciente = paciente;
    }

    //Métodos

    /* Método que busca los doctores que estén disponibles y también los doctores con los que el paciente
    asociado a esta historia haya tenido alguna cita */
    public ArrayList<Doctor> buscarCitaDoc(String especialidad, Hospital hospital) {
        ArrayList<Doctor> doctoresDisp = hospital.buscarTipoDoctor(especialidad);
        ArrayList<Doctor> docCita = new ArrayList<>();
        for (Doctor doc : doctoresDisp){
            for (Cita cita : historialCitas){
                if (doc.getCedula() == cita.getDoctor().getCedula()){
                    docCita.add(doc);
                }
            }
        }
        return docCita;
    }

    // Método que agrega las fórmulas médicas ingresadas como parámetro a listaFormulas
    public void agregarFormula(Formula formulaPaciente) {
        listaFormulas.add(formulaPaciente);
    }


    //Getter y setter
    public Paciente getPaciente() {
        return paciente;
    }

    public ArrayList<Formula> getListaFormulas() {
        return listaFormulas;
    }

    public void setListaFormulas(ArrayList<Formula> listaFormulas) {
        this.listaFormulas = listaFormulas;
    }

    public ArrayList<Cita> getHistorialCitas() {
        return historialCitas;
    }

    public void setHistorialCitas(ArrayList<Cita> historialCitas) {
        this.historialCitas = historialCitas;
    }

    public ArrayList<Enfermedad> getEnfermedades() {
        return enfermedades;
    }

    public void setEnfermedades(ArrayList<Enfermedad> enfermedades) {
        this.enfermedades = enfermedades;
    }

    public ArrayList<CitaVacuna> getHistorialVacunas() {
        return historialVacunas;
    }

    public void setHistorialVacunas(ArrayList<CitaVacuna> historialVacunas) {
        this.historialVacunas = historialVacunas;
    }
}

File: Python/src/gestor_aplicacion/administracion/historia_clinica.py
# Autores: Diego Andres Gracia Granados, Daniel Giraldo Vanegas, Elian David Velandia Riveros, Juan Camilo Gutierrez Martinez
# y Santiago Arboleda Acevedo
from src.gestor_aplicacion.personas.enfermedad import Enfermedad

# Clase destinada a crear historias clínicas a pacientes

class HistoriaClinica:
    # Atributos y constructor
    def __init__(self, paciente):
        self._PACIENTE = paciente
        self._lista_formulas = []
        self._historial_citas = []
        self._historial_vacunas = []
        self._enfermedades = []
    # Metodos
    # Método que busca los doctores que estén disponibles y también los doctores con los que el paciente
    # asociado a esta historia haya tenido alguna cita. Recibe un string y un objeto de clase hospital.
    # Devuelve una cita
    def buscar_cita(self, especialidad, hospital):
        disponibles = hospital.buscar_tipo_doctor(especialidad)
        doc_cita = []
        for doc in disponibles:
            for cita in self._historial_citas:
                if doc.cedula == cita.doctor.cedula:
                    doc_cita.append(doc)
        return doc_cita
    # Recibe un objeto de la clase Formula y la agrega al atributo
    def agregar_formula(self, formula):
        self._lista_formulas.append(formula)

    # Recibe un objeto de la clase enfermedad y la agrega al atributo

    def agregar_enfermedad(self, enfermedad):
        self._enfermedades.append(enfermedad)
    @property
    def enfermedades(self):
        return self._enfermedades

    @enfermedades.setter
    def enfermedades(self, value):
        self._enfermedades = value

    @property
    def lista_formulas(self):
        return self._lista_formulas

    @property
    def historial_citas(self):
        return self._historial_citas

    @property
    def historial_vacunas(self):
        return self._historial_vacunas
